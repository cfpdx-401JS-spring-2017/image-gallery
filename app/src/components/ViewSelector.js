import React, { Component } from 'react';
import PropTypes from 'prop-types';
import LoadingSpinner from './LoadingSpinner';
import List from './List';
import Thumbnail from './Thumbnail';
import Gallery from './Gallery';

class ViewSelector extends Component {
  constructor(props) {
    super(props);

    this.state = {
      CurrentView: List
    }

    this.handleListView = this.handleListView.bind(this);
    this.handleThumbnailView = this.handleThumbnailView.bind(this);
    this.handleGalleryView = this.handleGalleryView.bind(this);
  }

  static propTypes = {
    data: PropTypes.array.isRequired
  }

  handleListView() {
    this.setState({ CurrentView: List });
  }
  
  handleThumbnailView() {
    this.setState({ CurrentView: Thumbnail });
  }
  
  handleGalleryView() {
    this.setState({ CurrentView: Gallery });
  }

  render() {

    let { data, onDelete } = this.props;
    const { CurrentView } = this.state;

    if (!data) return <LoadingSpinner />

    return (
      <div>
        <div className="button-container">
          <button onClick={this.handleListView}>List</button>
          <button onClick={this.handleThumbnailView}>Thumbnail</button>
          <button onClick={this.handleGalleryView}>Gallery</button>
        </div>
        <CurrentView data={data}
                     onDelete={onDelete}
        />
      </div>
    );
  }
}

export default ViewSelector;