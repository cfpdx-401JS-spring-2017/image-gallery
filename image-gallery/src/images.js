import lordsOfDogtown from './photos/lordsOfDogtown.jpg';
import jayAdams from './photos/jayAdams.png';
import peggyOki from './photos/peggyOki.jpg';
import mikeKolar from './photos/mikeKolar.jpg';
import peggyTurner from './photos/peggyTurner.jpg';
import lauraThornhill from './photos/lauraThornhill2.jpg';
import rampage from './photos/rampage.jpg';
import shortId from 'shortid';
// let shortId = require('shortid');

const images = [
    {
        title: 'Z Boys',
        description: 'Zephyr skate team 1970\'s',
        url: lordsOfDogtown,
        _id: shortId.generate()
    },

    {
        title: 'Turn Only',
        description: 'Jay Adams gets low',
        url: jayAdams,
        _id: shortId.generate()
    },

    {
        title: 'Peggy Does Venice',
        description: 'Peggy Oki repping Zephr',
        url: peggyOki,
        _id: shortId.generate()
    },

    {
        title: 'Jumping Homies',
        description: 'Mike Kolar 1970\'s',
        url: mikeKolar,
        _id: shortId.generate()
    },

    {
        title: "Tippin' Toes",
        description: 'Laura Thornhill in skate competition 1970\'s',
        url: lauraThornhill,
        _id: shortId.generate()
    },

    {
        title: "Van Fans",
        description: 'Peggy Turner repping Vans in skate competition',
        url: peggyTurner,
        _id: shortId.generate()
    },

    {
        title: "Jumping Homies 2.0",
        description: 'Rampage Expo 1970\'s',
        url: rampage,
        _id: shortId.generate()
    }

];

export default {
    get() { return Promise.resolve(images) }
}