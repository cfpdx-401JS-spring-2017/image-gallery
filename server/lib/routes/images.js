const Router = require('express').Router;
const router = Router();
const Image = require('../models/image');

router

.get('/api/images',(res, req, next) => {
  Image.find()
    .then(images => res.send(images))
    .catch(next);
})

.get('/id', (req, res, next) => {
  const id = req.params.id;
  Image.findById(id)
    .then(image => res.send(image))
    .catch(next);
})

.post('/', (req, res, next) => {
  new Image(req.body)
    .save()
    .then(image => res.send(image))
    .catch(next);
})

.put('/id', (req, res, next) => {
  Image.findByIdAndUpdate(req.params.id, req.body, {new: true})
    .then(image => res.send(image))
    .catch(next);
})

.delete('./id', (req, res, next) => {
  Image.findByIdAndRemove(req.params.id)
    .then(response => {
      res.send({removed: response ? true : false });
    })
    .catch(next);
});

module.exports = router;
