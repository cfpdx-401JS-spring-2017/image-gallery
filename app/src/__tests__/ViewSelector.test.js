import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ViewSelector from '../components/ViewSelector';
import images from './_data';

describe('View Selector', () => {
  it('renders the default list view, along with buttons to change views', () => {
    const wrapper = shallow(
      <ViewSelector data={images} />
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('renders the thumbnail view', () => {
    const wrapper = shallow(
      <ViewSelector data={images} />
    );

    wrapper.find('button').at(1).simulate('click');  
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  
  it('renders the gallery view', () => {
    const wrapper = shallow(
      <ViewSelector data={images} />
    );

    wrapper.find('button').at(2).simulate('click');  
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});