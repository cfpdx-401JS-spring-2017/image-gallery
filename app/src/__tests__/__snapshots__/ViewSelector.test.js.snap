// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Selector renders the default list view, along with buttons to change views 1`] = `
<div>
  <div
    className="button-container"
  >
    <button
      onClick={[Function]}
    >
      List
    </button>
    <button
      onClick={[Function]}
    >
      Thumbnail
    </button>
    <button
      onClick={[Function]}
    >
      Gallery
    </button>
  </div>
  <List
    data={
      Array [
        Object {
          "description": "Seriously heart-melting black and tan bunny... OMG <3",
          "title": "Black & Tan",
          "url": "http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg",
        },
        Object {
          "description": "I don't know if this bunny fell and the picture was taken on impact, but I certainly prefer to think of it that way",
          "title": "Falling Bunny",
          "url": "http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg",
        },
        Object {
          "description": "Is it a rodent or a rabbit? I had to look it up to be sure. Then again, I'm still not sure",
          "title": "Marshmellow Puff",
          "url": "http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg",
        },
      ]
    }
  />
</div>
`;

exports[`View Selector renders the gallery view 1`] = `
<div>
  <div
    className="button-container"
  >
    <button
      onClick={[Function]}
    >
      List
    </button>
    <button
      onClick={[Function]}
    >
      Thumbnail
    </button>
    <button
      onClick={[Function]}
    >
      Gallery
    </button>
  </div>
  <Gallery
    data={
      Array [
        Object {
          "description": "Seriously heart-melting black and tan bunny... OMG <3",
          "title": "Black & Tan",
          "url": "http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg",
        },
        Object {
          "description": "I don't know if this bunny fell and the picture was taken on impact, but I certainly prefer to think of it that way",
          "title": "Falling Bunny",
          "url": "http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg",
        },
        Object {
          "description": "Is it a rodent or a rabbit? I had to look it up to be sure. Then again, I'm still not sure",
          "title": "Marshmellow Puff",
          "url": "http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg",
        },
      ]
    }
  />
</div>
`;

exports[`View Selector renders the thumbnail view 1`] = `
<div>
  <div
    className="button-container"
  >
    <button
      onClick={[Function]}
    >
      List
    </button>
    <button
      onClick={[Function]}
    >
      Thumbnail
    </button>
    <button
      onClick={[Function]}
    >
      Gallery
    </button>
  </div>
  <Thumbnail
    data={
      Array [
        Object {
          "description": "Seriously heart-melting black and tan bunny... OMG <3",
          "title": "Black & Tan",
          "url": "http://f.cl.ly/items/3g3J1G0w122M360w380O/3726490195_f7cc75d377_o.jpg",
        },
        Object {
          "description": "I don't know if this bunny fell and the picture was taken on impact, but I certainly prefer to think of it that way",
          "title": "Falling Bunny",
          "url": "http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-25__605.jpg",
        },
        Object {
          "description": "Is it a rodent or a rabbit? I had to look it up to be sure. Then again, I'm still not sure",
          "title": "Marshmellow Puff",
          "url": "http://static.boredpanda.com/blog/wp-content/uploads/2015/09/cute-bunnies-110__605.jpg",
        },
      ]
    }
  />
</div>
`;
